### PR TITLE
Psykal part 2

### PR DESCRIPTION
**Pull request by rupertford**
I think I've addressed all of the issues raised in issue #4 so we are ready for review.

### CODE REVIEW

File: src/psykal/mydomain_kern.f90
Comment (arporter) [inline src/psykal/mydomain_kern.f90]: `zbtr` converted into a kernel-local scalar.

File: src/psykal/psy.f90
Comment (arporter) [inline src/psykal/psy.f90]: Slightly counter-intuitive argument ordering here but will let it go since this routine is a temporary work-around prior to us getting the infrastructure up and running.

File: src/psykal/psy.f90
Comment (arporter) [inline src/psykal/psy.f90]: For double precision this should be 0.d0.

Comment (arporter) [reply src/psykal/psy.f90]: Oh, and in NEMO, the convention is that jk=1 is the top layer (surface) while jk=jpk is the bottom layer. This is a model-dependent choice (although I think it is a common one). Perhaps we need meta-data for this?
Therefore, this routine should be named `zero_bottom_layer`.

File: src/psykal/psy.f90
Comment (arporter) [inline src/psykal/psy.f90]: Again, should be 0.d0 for double precision.

File: src/psykal/psy.f90
Comment (arporter) [inline src/psykal/psy.f90]: With NEMO's layer-indexing convention, this routine should actually be `zero_top_layer`.

File: src/psykal/psy.f90
Comment (arporter) [inline src/psykal/psy.f90]: Again, this should be named `multiply_top_layer`

File: src/psykal/zwx2_kern.f90
Comment (arporter) [inline src/psykal/zwx2_kern.f90]: Probably better to use 1.d0

Comment (arporter) [reply src/psykal/zwx2_kern.f90]: ...for both since `zdt` is also real and not integer

File: src/psykal/zwx2_kern.f90
Comment (arporter) [inline src/psykal/zwx2_kern.f90:24]: Indexing changes so kernel writes to `(ji,jj,jk)` rather than `(ji,jj,jk+1)` - good.

File: src/psykal/zwxy2_kern.f90
Comment (arporter) [inline src/psykal/zwxy2_kern.f90]: Should be 1.d0.

File: src/psykal/psy.f90
Comment (arporter) [inline src/psykal/psy.f90]: I failed to notice this when reviewing #3 but I think that the ordering of the `jk, jj, ji` kernel arguments is counter-intuitive. Please can you change this to `ji, jj, jk` for all kernels.

**Review by arporter — CHANGES_REQUESTED** [review]
Generally good. My main issue is with the naming of the 'top' and 'bottom' layers - this is currently round the wrong way. There is also the matter of the ordering of the array-index arguments passed to kernels - I think it should be `ji,jj,jk` and not `jk,jj,ji` as it is now.
Please also address the other more minor issues (such as use of 1.d0 rather than 1 or 1.e0).

File: src/psykal/psy.f90
Comment (rupertford) [reply src/psykal/psy.f90]: I've changed this anyway for consistency.

File: src/psykal/psy.f90
Comment (rupertford) [reply src/psykal/psy.f90]: Swapped top and bottom layer names and code.

File: src/psykal/psy.f90
Comment (rupertford) [reply src/psykal/psy.f90]: Done.

File: src/psykal/psy.f90
Comment (rupertford) [reply src/psykal/psy.f90]: Done.

File: src/psykal/zwx2_kern.f90
Comment (rupertford) [reply src/psykal/zwx2_kern.f90]: Done

File: src/psykal/psy.f90
Comment (rupertford) [reply src/psykal/psy.f90]: Done.

File: src/psykal/psy.f90
Comment (rupertford) [reply src/psykal/psy.f90]: Double precision change done.

File: src/psykal/zwxy2_kern.f90
Comment (rupertford) [reply src/psykal/zwxy2_kern.f90]: Done.

File: src/psykal/psy.f90
Comment (rupertford) [reply src/psykal/psy.f90]: Changed to ji,jj,jk as requested. I agree it looks better.

**Comment by rupertford** [comment]
OK, ready for second review.

**Comment by arporter** [comment]
Changes all look good now. I'll hold off approving though until I've built and run the code.

**Review by arporter — APPROVED** [review]

**Comment by arporter** [comment]
Built and run fine and answers agree with original.
All done.

**Pull request merged**
